src/__tests__/Authorization.test.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { screen, render, waitFor } from '@testing-library/react';
+import { Authorization } from '../components/Authorization';
+import { FusionAuthProvider } from '../providers/FusionAuthProvider';
+import { FusionAuthLogoutButton } from '../components/FusionAuthLogoutButton';
+import axios from 'axios';
+
+let location: Location;
+describe('Authorization Component', () => {
+    beforeEach(() => {
+        location = window.location;
+        jest.spyOn(window, 'location', 'get').mockRestore();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('Authorization Component does not render children when no user is present and no role is passed', async () => {
+        await renderProvider();
+
+        expect(screen.queryByText('Logout')).toBeNull();
+    });
+
+    test('Authorization Component does not render children when user is not present', async () => {
+        await renderProvider('admin');
+
+        expect(screen.queryByText('Logout')).toBeNull();
+    });
+
+    test('Authorization Component renders children when user is present with the correct role', async () => {
+        const mockedLocation = {
+            ...location,
+            search: 'http://localhost?code=here&state=00000000000000000000000000000000000000000000000000000000&error=none',
```

Comment:
Overall these tests look great!

One suggestion: it looks like the "search" is the same for all of these (or maybe it isn't). Might be good to define them as a `const` outside of the tests, in case we need to change them. That constant name would also tell us the meaning of that particular URL.